style-test.py

```diff
@@ -0,0 +1,148 @@
+    shutil.copytree(src, dest)
+
+    with open("style-test.txt", "r") as file:
+        tests = file.readlines()
+
+    with open(dest_prc, "r") as in_file:
+        buf = in_file.readlines()
+
+    pos = len(buf)-2
+
+    # find first occurence of style-guide checks in proselintrc
+    for index, line in enumerate(buf):
+        if line.startswith("        \"style-guide."):
+            pos = index
+            break      
+
+    # copy over new checks to proselintrc from style-text.txt
+    with open(dest_prc, "w") as out_file:
+        for index, line in enumerate(buf):
+            if index == pos-1 and not line.endswith(",\n"):
+                line = line[0:line.rfind("\n")] + ",\n"
+            if index < pos:
+                out_file.write(line)
+        for test in tests:
+            out_file.write(test)                
+        out_file.write("    }\n")
+        out_file.write("}")
+
+
+def remove_ignored_lines(txt):
+    """Function to remove ignored lines from text"""
+    index = 0
+    while index<len(txt):
+        if "startignore" in txt[index]:
+            while index<len(txt) and "endignore" not in txt[index]:
+                txt[index]="ignore_line\n"
+                index = index+1
+        index = index+1 
+
+    f= open("temp.txt","w+")
+    for line in txt:
+        f.write(line)
+
+    txt = open("temp.txt", "r")
+    os.remove("temp.txt")
+
+    return txt            
+
+
+def run_checks(paths):
+
+    """Function to run checks on the docs."""
+    file_path = os.path.realpath(__file__)
+
+    # find path for all .rst files
+    search_path = file_path[0:file_path.rfind('/')]
+
+    # Make a list of paths to check for
+    path_list = []
+
+    # Add paths if provided by user
+    if paths:
+        search_path = search_path + "/"
+        path_list = [search_path + path for path in paths]
+
+    # Add all rst files if specific paths not provided by user
+    else:
+        for filename in glob.glob(os.path.join(search_path, '*.rst')):
+            path_list.append(filename)
+
+    for filename in path_list:        
+        
+        # read the file 
+        with open(filename, "r") as file:
+            txt = file.readlines()
+
+        # remove ignored lines
+        txt = remove_ignored_lines(txt)
+
+        # lint the text
+        errors = proselint.tools.lint(txt)
+        err_list = []
+
+        for e in errors:
+
+            # ignore tests for curly quotes
+            if e[0] == "typography.symbols.curly_quotes":
+                continue
+            
+            # prepare error message
+            check = "check: %s, " %e[0] 
+            msg = "message: %s, " %e[1]
+            line = "line: %d, " %(1 + e[2])
+            column = "column: %d, " %(1 + e[3])
+            start = "start: %d, " %(1 + e[4])
+            end = "end: %d, " %(1 + e[5])
+            extent = "extent: %d, " %e[6]
+            severity = "severity: %s, " %e[7]
+            replace = "replacements: %s " %e[8]
+
```

Comment:
In general, I don't think it is a good practice to compose strings in several places.
Here you are pulling unnamed strings out of `e`, and concat-ing then with their name, and then again later concat-ing them together. This requires the next person who has to work on this code to figure out what is going in a few different places. For a simple script like this, on our small team, that's not a huge deal. But in a larger production environment, this is a habit that leads to technical debt.

The other thing is that pulling `e` into named variables (probably a single dict) but without the additional text is that it makes it easier to change what the whole output string looks like. -- (Very relevant, since I'm suggesting you do exactly that in my comment below.)

```
for e in errors:

    check = e[0]
    msg = e[1]
    line = e[2] + 1   # just a matter of convention, this is better than  1 + e[2]

```

This just makes it clearer what you are doing.

Anyway... 
All this error-message code would be cleaner if we upgraded to python 3.6 and used f-strings.

But I don't think we should force that on people. 

I wonder if there is a simple templating package that would replicate the function of f-strings so that you could build the whole error msg in one place:

```
err_msg = """
 {{file_name_}} | ln {{linum}} | col {{colnum}} \n
{{text_excerpt}} \n
{{message}} \n
{{check}} | {{severity}}
"""

(See my other comment for why that format...)

This would make the code more readable and maintainable.